tests/test_pytypes.py

```diff
@@ -488,3 +488,45 @@ def test_builtin_functions():
         "object of type 'generator' has no len()",
         "'generator' has no length",
     ]  # PyPy
+
+
+def test_weakref():
+    from weakref import getweakrefcount
+
+    # Apparently, you cannot weakly reference an object()
+    class WeaklyReferenced(object):
+        pass
+
+    called = [0]
+
+    def callback(wr):
+        # No `nonlocal` in Python 2
+        called[0] += 1
+
+    obj = WeaklyReferenced()
+    assert getweakrefcount(obj) == 0
+    wr = m.weakref_from_handle(obj)  # noqa: F841
+    assert getweakrefcount(obj) == 1
+
+    obj = WeaklyReferenced()
+    assert getweakrefcount(obj) == 0
+    wr = m.weakref_from_handle_and_function(obj, callback)  # noqa: F841
```

Comment:
Can't you just not assign the weak reference?

Reply:
Well, it will be gone. Will the destructor still be called then? I don't think so. I think I/we want to check this callback being called on `del obj`, no?

Reply:
This is in the interactive repl, though?

Have a look at this:

```
$ python3 - << EOF
> import weakref
> class c: pass
> o = c()
> weakref.ref(o, lambda x: print(1))
> print('bump')
> del o
> 
> EOF
bump
```
vs.
```
$ python3 - << EOF
> import weakref
> class c: pass
> o = c()
> wr = weakref.ref(o, lambda x: print(1))
> print('bump')
> del o
> EOF
bump
1
```